code/script.js

```diff
@@ -218,108 +224,118 @@ const generateBoard = () => {
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
+      alert(`No, the person isn't a ${value}! Remove all with cigarettes.`)
     }
+  } else if (category === "hair") {
+    if (keep) {
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
+      alert(`Yes, the person has ${value}! We're keeping everyone with ${value}.`);
+    } else {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
+      alert(`Nope, the person doesn't have ${value}! We're getting rid of everyone with ${value}.`);
+    }
+  } else if (category === "eyes") {
+    if(keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
+      alert(`Yes, the person has ${value} eyes! We're keeping anybody with ${value} eyes.`);
+    } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
+      alert(`Sorry, the person does not have ${value} eyes! Removing everyone with ${value} eyes.`);
   }
+  }
+};
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
 
   // Invoke a function to redraw the board with the remaining people.
-}
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+const guess = (chosenPerson) => { 
+  const timeToGuess = confirm(`Are you sure you want to guess on ${chosenPerson}?`);
+  if (timeToGuess) {
+    checkMyGuess(chosenPerson);
+  }
+};
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+const checkMyGuess = (chosenPerson) => {
+  if (chosenPerson === secret.name) {
+    winOrLoseText.innerHTML = `CONGRATULATIONS! <br> 
+    YOU PICKED THE RIGHT ONE!`;
+  } else {
+    winOrLoseText.innerHTML = `Aaah dang it, that was the wrong one!`;
```

Comment:
Your language makes the game interact with the player in a fun way! Playable english!